src/App.php

```diff
@@ -49,15 +50,23 @@
 class App extends AbstractApp {
 
 	/**
-	 * @var array Languages supported by the app
-	 */
-	public $supportedLanguages = [ 'en', 'fr' ];
-
-	/**
-	 * @return array Languages supported by the app
+	 * Get codes of all langages supported by CopyPatrol. This list is retrieved from the
+	 * copyright_diffs database table, and cached for a week.
+	 * @return string[] The 2-letter language codes.
 	 */
 	public function getSupportedLanguages() {
-		return $this->supportedLanguages;
+		// Use the cached list if possible.
+		$cacheItem = $this->slim->cache->getItem( 'supported-languages' );
+		if ( $cacheItem->isHit() ) {
+			return $cacheItem->get();
+		}
+
+		// Otherwise get the list from the database, and cache it for a week.
+		$langs = $this->getPlagiabotDao()->getLanguages();
+		$cacheItem->expiresAfter( new DateInterval( 'P7D' ) );
```

Comment:
It's the [interval_spec](http://php.net/manual/en/dateinterval.construct.php), quite redundant really, I guess their aiming at future compatibility with things. I think it's also something to do with ISO8601 (with the 'T' being a delimiter for the time component)....

>  The format starts with the letter P, for "period." Each duration period is represented by an integer value followed by a period designator. If the duration contains time elements, that portion of the specification is preceded by the letter T. 

I'll make a comment about the format meaning.